### PR TITLE
Forward thinking naming convention

### PR DESCRIPTION
**Pull request by SCdF**
Convert to a more forward-thinking naming convention with a namespace:

```
foo:bar:1.2.3-beta.4
```

Means application `foo`, ddoc `bar`, version `1.2.3-beta.4`.

WIP. PouchDB tests don't pass because PouchDB doesn't support CommonJS in views. Instead, convert to an int. tests and re-write to use CouchDB instead.

### CODE REVIEW

**Comment by SCdF** [comment]
NB: @garethbowen don't merge, I'll deal with it, need to do everything at once I think (ie alongside api / webapp)

**Review by garethbowen — APPROVED** [review]

File: .travis.yml
Comment (mandric) [inline .travis.yml]: 2 sleeps?

File: README.md
Comment (mandric) [inline README.md:4]: I like line breaks.

File: README.md
Comment (mandric) [inline README.md:11]: I would rather decouple the ddoc ID from the database name and make this an argument when you call the medic-api endpoint.  Simplifies a few things imo.

**Review by mandric — CHANGES_REQUESTED** [review]
Requesting some discussion here because I still don't understand the purpose of this odd naming convention that we're writing a bunch of code around.  I'd like to simply the API a bit if possible https://github.com/medic/medic-api/pull/187/files#r134029805

File: README.md
Comment (SCdF) [reply README.md:11]: > and make this an argument 

What does this refer to here? If this refers to the db name, then we're not locking that down in this naming convention, you can (in theory) deploy to any db name you like. It's just hard-coded in api for now because I want to do one thing at time (see my comment on that PR).

If this refers to the ddoc name then, well, I don't know about other applications based in couchdb, but ours has our ddoc name hardcoded in a thousand different places, so you couldn't deploy that to another name even if you wanted to.

Comment (mandric) [reply README.md:11]: Even if the ddoc name is hardcoded in lots of places I would try to avoid hard coding it in any new code.  We might want to break up our ddoc(s) more or deploy them as separate apps one day in the future.  Since this is 3.0 code I thought I would mention it.

Comment (garethbowen) [reply README.md:11]: I don't follow this point. Firstly it's entirely appropriate to hardcode the ddoc name. If we decide to split things up or change the name we will hardcode the new name.

Secondly there's nothing here that talks about the db name so there's no coupling between ddoc and db names.

I think using the term ddoc name is a little confusing - this is the primary ddoc (which I've been calling the application). It bundles the api and sentinel modules, as well as our medic-client and medic-conflicts ddocs. Nothing is stopping us from breaking out more ddocs in the future and bundling them together into one file for deployment.

This piece of documentation is just talking about each app in the builds db having a unique ID and a version. The  unique ID for our main app is `medic:medic`. We could also ship `medic:reporter` and `medic:couch2pg` and anything else we want at a later date.

**Review by garethbowen — APPROVED** [review]
Nice one!

File: README.md
Comment (mandric) [reply README.md:11]: My main point here is just that the db and app name are both used (mapped from the build db doc name) when horti installs a build.  It's quite possible I'm misunderstanding something and it would be nice to review this stuff together since lots of moving parts here but I think my point is better illustrated on the other PR: https://github.com/medic/medic-api/pull/187/files#r134029805

Comment (garethbowen) [reply README.md:11]: Unless I'm mistaken the DB name isn't defined in the built bundle. We should be able to build the medic app and deploy to any database. This PR and the entire medic-builds-repo should not care what the name of the local database is.

Comment (SCdF) [reply README.md:11]: I cannot stress enough that this is just a naming scheme. It may be hard to tell because everything else is hardcoded behind it, because I do not see the value in spending large amounts of time architecting and then building something we aren't sure about and don't currently need. My priority is to get this merged so we can start generating builds, and people can start testing branches using it, and we can start getting actual experience in this style of deployment. I have no doubt that large amounts of things will change between now and us being international deployment superstars. But, until we merge developers cannot branch off master and have their branches pushed to horti, and have people test their branches with horti, etc.

Anyway, on to the points as I read them:

The database it gets deployed to is not hard-coded into this scheme. It's not even mentioned in this readme. It is only hard-coded in the backend code because of the aforementioned YAGNI.

The ddoc name is mapped to the application name. However, if we ever feel like allowing this to not be true, we can just say that the ddoc is the application name *by default*, but you can either specify a different name in the metadata associated with the build[1], or pass a different one when you install it. Again, we don't support this because it asks a dozen questions of us we don't need to answer right now. I'm also not convinced of the merit of it (it doesn't have any value inside of medic afaict), so it's very low in the priority list, around when(/if) we support third parties deploying in this fashion.

It's definitely something we can consider when we are at that stage.

[1] We are currently using `build_time` from Kanso. When/If we drop Kanso or when/if we want other people to use this formally we can have our own metadata section.

File: .travis.yml
Comment (SCdF) [reply .travis.yml]: Copy paste fail!

File: README.md
Comment (SCdF) [reply README.md:4]: Me too!

Comment (alxndrsn) [reply README.md:4]: +1.  I'd like to take this moment to complain about https://github.com/medic/horticulturalist/pull/1/files#diff-1fdf421c05c1140f6d71444ea2b27638L69

**Pull request merged**
